File: OJ/leetcode/2079/Solution.java
class Solution {
    public int wateringPlants(int[] plants, int capacity) {
        int n = plants.length;
        int ans = 0;
        int currentCap = capacity;
        for (int i = 0; i < n; i++) {
            if (currentCap < plants[i]) {
                currentCap = capacity;
                // refill water can and walk to (i - 1)
                ans += i * 2;
            }
            currentCap -= plants[i];
            ans++;
        }
        return ans;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        int[] plants = {1,1,1,4,2,3};
        int capacity = 4; // ans = 30
        System.out.println("test: " + sol.wateringPlants(plants, capacity));
    }
}

File: OJ/leetcode/2079/Solution.py
from typing import List

class Solution:
    def wateringPlants(self, plants: List[int], capacity: int) -> int:
        ans = 0
        currentCap = capacity
        for i, require in enumerate(plants):
            if (currentCap < require) :
                # refill water can and walk to (i - 1)
                currentCap = capacity
                ans +=2 * i
            currentCap -= require
            ans += 1
        return ans

if __name__ == '__main__':
    sol = Solution()
    plants = [1,1,1,4,2,3]
    capacity = 4 # ans = 30
    print("test:", sol.wateringPlants(plants, capacity))